server/model/status_page.js

```diff
@@ -98,6 +145,40 @@ class StatusPage extends BeanModel {
         return $.root().html();
     }
 
+    /**
+     * Get all data required for RSS
+     * @param {StatusPage} statusPage Status page to get data for
+     * @returns {object} Status page data
+     */
+    static async getRSSPageData(statusPage) {
+        // get all heartbeats that correspond to this statusPage
+        const config = await statusPage.toPublicJSON();
+
+        // Public Group List
+        const showTags = !!statusPage.show_tags;
+
+        const list = await R.find("group", " public = 1 AND status_page_id = ? ORDER BY weight ", [
+            statusPage.id
+        ]);
+
+        let heartbeats = [];
+
+        for (let groupBean of list) {
+            let monitorGroup = await groupBean.toPublicJSON(showTags, config?.showCertificateExpiry);
+            for (const a of monitorGroup.monitorList) {
+                const id = a.id;
+                const heartbeat = await R.findOne("heartbeat", "monitor_id = ? ORDER BY time DESC", [ id ]);
+                if (heartbeat && !heartbeat.status) {
+                    heartbeats.push({ ...a, status: heartbeat.status, time: heartbeat.time });
```

Comment:
nit: can be simplified

```suggestion
            for (const monitor of monitorGroup.monitorList) {
                const heartbeat = await R.findOne("heartbeat", "monitor_id = ? ORDER BY time DESC", [ monitor.id ]);
                if (heartbeat && !heartbeat.status) {
                    heartbeats.push({
                        ...monitor,
                        status: heartbeat.status,
                        time: heartbeat.time,
                    });
```

Reply:
Fixed in chore commit